share/spice/sales_tax_holiday/sales_tax_holiday.js

```diff
@@ -0,0 +1,70 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_sales_tax_holiday = function(api_result){
+        // Error checking
+        var errorMessage = "UNKNOWN ERROR";
+        var isError = false;
+        if (!api_result 
+         || (typeof api_result.header === 'undefined')
+         || api_result.header.status !== "SUCCESS" 
+         || !api_result.taxHolidays
+         || (typeof api_result.taxHolidays === 'undefined')
```

Comment:
Though I have tested this API end point for all possible failures. But I am little orthodox in putting all checks before I call spice.add(..). Plus javascript, I am not expert. 
I am putting API response in case of state not found. as example.
`
{
  "header" : {
    "status" : "FAIL",
    "errorCode" : "NOT_FOUND",
    "errorMessage" : "No tax holidays found for state NJ",
    "apiVersion" : "0.0.1",
    "serverResponseTime" : "1.039 msecs",
    "request" : {
      "requestParams" : [ {
        "key" : "request_id",
        "value" : "123"
      }, {
        "key" : "state",
        "value" : "NJ"
      } ]
    }
  },
  "taxHolidays" : [ ]
}
`